Compiling Error Since the Use of InferShapeContextBase in decayed_adagrad_op

Compile failed since the use `InferShapeContextBase` in decayed_adagrad_op. It seems that `InferShapeContextBase` had been removed.

```
/Users/baidu/Documents/codeRep/paddle-master/Paddle/paddle/operators/decayed_adagrad_op.cc:25:30: error: no type named 'InferShapeContextBase' in namespace 'paddle::framework'; did you mean 'InferShapeContext'?
  void InferShape(framework::InferShapeContextBase *ctx) const override {
                  ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
                             InferShapeContext
/Users/baidu/Documents/codeRep/paddle-master/Paddle/paddle/framework/shape_inference.h:25:7: note: 'InferShapeContext' declared here
class InferShapeContext {
      ^
1 error generated.
make[2]: *** [paddle/operators/CMakeFiles/decayed_adagrad_op.dir/decayed_adagrad_op.cc.o] Error 1
```

--- a/paddle/operators/decayed_adagrad_op.cc
+++ b/paddle/operators/decayed_adagrad_op.cc
@@ -22,7 +22,7 @@ class DecayedAdagradOp : public framework::OperatorWithKernel {
   using framework::OperatorWithKernel::OperatorWithKernel;
 
  protected:
-  void InferShape(framework::InferShapeContextBase *ctx) const override {
+  void InferShape(framework::InferShapeContext *ctx) const override {
     PADDLE_ENFORCE(ctx->HasInput("Param"),
                    "Input(Param) of DecayedAdagradOp should not be null.");
     PADDLE_ENFORCE(ctx->HasInput("Grad"),

